ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteSessionImpl.java

```diff
@@ -0,0 +1,88 @@
+package org.hibernate.ogm.datastore.ignite.impl;
+
+import java.util.List;
+
+import org.hibernate.Criteria;
+import org.hibernate.HibernateException;
+import org.hibernate.event.spi.EventSource;
+import org.hibernate.internal.CriteriaImpl;
+import org.hibernate.ogm.OgmSessionFactory;
+import org.hibernate.ogm.datastore.ignite.loader.criteria.OgmCriteriaLoader;
+import org.hibernate.ogm.datastore.ignite.loader.criteria.impl.CriteriaCustomQuery;
+import org.hibernate.ogm.exception.NotSupportedException;
+import org.hibernate.ogm.hibernatecore.impl.OgmSessionImpl;
+import org.hibernate.resource.transaction.TransactionCoordinator;
+import org.hibernate.resource.transaction.backend.jta.internal.JtaTransactionCoordinatorImpl;
+
+public class IgniteSessionImpl extends OgmSessionImpl {
```

Comment:
Hi!

We have to do this because we implements Criteria API, but
in OgmSessionImpl we have only

 @Override public Criteria createCriteria(Class persistentClass, String
alias) { //TODO plug the Lucene engine throw new NotSupportedException(
"OGM-23", "Criteria queries are not supported yet" ); }
And I've always wanted to ask why don't you support criteria queries?

Victor


2016-03-08 2:34 GMT+03:00 DavideD <notifications@github.com>:

> In
> ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteSessionImpl.java
> <https://github.com/hibernate/hibernate-ogm/pull/628#discussion_r55292926>
> :
>
> > +
> > +import java.util.List;
> > +
> > +import org.hibernate.Criteria;
> > +import org.hibernate.HibernateException;
> > +import org.hibernate.event.spi.EventSource;
> > +import org.hibernate.internal.CriteriaImpl;
> > +import org.hibernate.ogm.OgmSessionFactory;
> > +import org.hibernate.ogm.datastore.ignite.loader.criteria.OgmCriteriaLoader;
> > +import org.hibernate.ogm.datastore.ignite.loader.criteria.impl.CriteriaCustomQuery;
> > +import org.hibernate.ogm.exception.NotSupportedException;
> > +import org.hibernate.ogm.hibernatecore.impl.OgmSessionImpl;
> > +import org.hibernate.resource.transaction.TransactionCoordinator;
> > +import org.hibernate.resource.transaction.backend.jta.internal.JtaTransactionCoordinatorImpl;
> > +
> > +public class IgniteSessionImpl extends OgmSessionImpl {
>
> What's the reason for creating a specific session for Ignite?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hibernate/hibernate-ogm/pull/628/files#r55292926>.
>